### DISCUSSION TITLE
Question about shuffle

### DISCUSSION THREAD

**Discussion by jinz2014** [intel/llvm#3181] [Q&A]
https://manpages.debian.org/testing/opencl-1.2-man-doc/shuffle.3clc.en.html

Does a user need to implement OpenCL shuffle in SYCL ?

**Comment by alexbatashev** [comment]
@jinz2014 have you considered `vec` swizzles (4.10.2.3 from [spec](https://www.khronos.org/registry/SYCL/specs/sycl-1.2.1.pdf))? They seem to have the same purpose. The only downside I see is that mask must be known at compile time. 

Otherwise, you're right, there's no `shuffle` builtin in SYCL.
